src/pages/patientView/SampleManager.tsx

```diff
@@ -279,6 +292,53 @@ class SampleManager {
         }
         return '';
     }
+
+    getClinicalAttributeList(samples: Array<ClinicalDataBySampleId>) {
+        let clinicalAttributes: { [id: string]: ClinicalAttribute } = {};
+        let output: { id: string; value: string }[] = [];
+        samples.forEach((sample, sampleIndex) => {
+            sample.clinicalData.forEach((clinicalData, clinicalDataIndex) => {
+                if (
+                    clinicalAttributes[clinicalData.clinicalAttributeId] ===
+                    undefined
+                ) {
+                    output.push({
+                        id: clinicalData.clinicalAttributeId,
+                        value: clinicalData.clinicalAttribute.displayName,
+                    });
+                    clinicalAttributes[clinicalData.clinicalAttributeId] =
+                        clinicalData.clinicalAttribute;
+                }
+            });
+        });
+        return output;
+    }
+
+    getClinicalValueToSamplesMap(
+        samples: Array<ClinicalDataBySampleId>,
+        clinicalAttributeId: string
+    ) {
+        let clinicalAttributeSamplesMap = new Map();
```

Comment:
@adamabeshouse, by using untyped Map() here aren't we sacrificing typing?  There's probably a way to improve with generic, right?